### PR TITLE
Make NAT gateway generation prefix a variable

### PR DESCRIPTION
**Pull request by SkalaNetworks**
This makes the prefix in front of Pods/STS generated for NAT gateways a configurable variable from the ConfigMap.

### CODE REVIEW

File: pkg/util/vpc_nat_gateway.go [hunk 1]
```diff
@@ -2,10 +2,17 @@ package util
 
 import "fmt"
 
+var (
```
Comment (bobz965) [inline pkg/util/vpc_nat_gateway.go:5]: how about using a const?

Comment (SkalaNetworks) [reply pkg/util/vpc_nat_gateway.go:5]: Allows for changing the prefix on the fly from other packages, such as the one managing the configmap for the nat gateway

Comment (bobz965) [reply pkg/util/vpc_nat_gateway.go:5]: ok

**Review by bobz965 — APPROVED** [review]

**Comment by bobz965** [comment]
@SkalaNetworks please fix： 
image

File: pkg/controller/vpc_nat.go [hunk 2]
```diff
@@ -22,6 +22,13 @@ func (c *Controller) resyncVpcNatConfig() {
 		return
 	}
 
+	// Prefix used to generate the name of the StatefulSet/Pods for a NAT gateway
+	// By default it is equal to the value contained in 'util.VpcNatGwNamePrefix'
+	vpcNatGwNamePrefix := cm.Data["natGwNamePrefix"]
+	if vpcNatGwNamePrefix != "" {
+		util.VpcNatGwNamePrefix = vpcNatGwNamePrefix
+	}
```
Comment (zhangzujian) [inline pkg/controller/vpc_nat.go:30]: What if `natGwNamePrefix` was set back to empty?

Comment (SkalaNetworks) [reply pkg/controller/vpc_nat.go:30]: Bad things happen I guess :) I'll fix that

**Comment by SkalaNetworks** [comment]
Hey, sorry I had a lot of stuff on my plates, just fixed this so the prefix resets back to default if the value is changed to default.

Note that changing a prefix AFTER NAT gateways have been provisioned is a destructive operation as Kube-OVN will fail to track the old gateways (it uses labels/names, and they don't match the new prefix anymore). I'll document that in the docs if this is merged.

**Review by zhangzujian — APPROVED** [review]

**Pull request merged**